### PR TITLE
build: add armv7l-musl and aarch64-darwin builds

### LINKED ISSUE #1736: Rasbperry 3 (armv7l-unknown-linux-musl)
#### Is your feature request related to a problem? Please describe.
Install starship in raspberry pi 3

#### Describe the solution you'd like
When I run the installation script in a rasbian system based on armv7l-unknown-linux-musl starship should be installed normaly

#### Describe alternatives you've considered
May be to have an installation script for raspbian

### LINKED ISSUE #1901: Release a build for aarch64-apple-darwin (Apple M1)
#### Is your feature request related to a problem? Please describe.

With the release of M1 Macbooks and Mac Minis, we need to add `aarch64-apple-darwin` to our release pipeline.

### PR DESCRIPTION
**Pull request by matchai**
#### Description

Build releases for Apple M1 and Raspberry Pi 3.

#### Motivation and Context

Closes #1736
Closes #1901 

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

**Comment by davidkna** [comment]
On linux [cross](https://github.com/rust-embedded/cross) works very well for cross-compiling to arm without any extra setup. It should also be able to simplify the musl builds.

Without cross, armv7l-musl requires a gcc build that both cross-compiles to armv7l and uses musl libc, which might not be shipped with ubuntu. clang might get rid of this requirement, but  I think an arm build of musl is still required.

Both aarch64-darwin and armv7l-musl could also require linker configuration fro cross-compiling, e.g.:
```toml
# .cargo/config.toml
[target.armv7l-unknown-linux-musl]
# --target=armv7l-unknown-linux-musl might be better off in rustflags as a linker flag
linker = "clang --target=armv7l-unknown-linux-musl"
```

For newer raspberry pis I'd also like to see an `aarch64-unknown-linux-musl` build.
The default [Raspberry Pi OS](https://www.raspberrypi.org/software/operating-systems/) by the raspberry foundation for the pi has hf in the name, so for the pi an `arm-unknown-linux-musleabihf` build might be better suited than `armv7l-unknown-linux-musl`.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -80,15 +82,15 @@ jobs:
           target: ${{ matrix.target }}
 
       - name: Setup | musl tools
-        if: matrix.target == 'x86_64-unknown-linux-musl'
+        if: endsWith( matrix.target, 'musl' )
         run: sudo apt install -y musl-tools
 
       - name: Build | Build
-        if: matrix.target != 'x86_64-unknown-linux-musl'
+        if: endsWith( matrix.target, 'musl' )
```
Comment (mentos1386) [inline .github/workflows/deploy.yml:89]: Isn’t this supposed to be negated?  As in, `not endsWith(matrix.target, ‘musl’)`, not sure about the syntax though.

**Comment by mentos1386** [comment]
@davidkna @matchai I would really like to use this on my armv7 machine. What needs to be done/fixed for this to be merged? I’m not a rust-dev, but could take a look to make it work.

**Comment by matchai** [comment]
@mentos1386 - While we work on getting a binary released, you are free to build it on your armv7 machine.
With the latest version of rust installed, running `cargo install starship` will build and install the latest release on your machine.

**Comment by imabuddha** [comment]
As someone who uses a Pi 4 (currently with 32bit OS, but moving to 64bit as soon as it's out of beta) I'm very happy to see official support is in the works. Meanwhile I'll give @matchai's suggestion a try.

**Pull request closed without merging**